pyracf/common/security_admin.py

```diff
@@ -157,32 +179,57 @@ def _make_request(
                 security_request.dump_request_xml(encoding="utf-8"),
                 secret_traits=self.__secret_traits,
             )
+        request_xml = self.__logger.redact_request_xml(
+            security_request.dump_request_xml(encoding="utf-8"),
+            secret_traits=self.__secret_traits,
+        )
         if self._generate_requests_only:
-            request_xml = self.__logger.redact_request_xml(
-                security_request.dump_request_xml(encoding="utf-8"),
-                secret_traits=self.__secret_traits,
-            )
             self.__clear_state(security_request)
             return request_xml
         result_xml = self.__logger.redact_result_xml(
             self.__irrsmo00.call_racf(
-                security_request.dump_request_xml(), irrsmo00_precheck
+                security_request.dump_request_xml(),
+                irrsmo00_precheck,
+                self.__running_userid,
             ),
             self.__secret_traits,
         )
         self.__clear_state(security_request)
+        if isinstance(result_xml, list):
+            # When IRRSMO00 encounters some errors, it returns no XML response string.
+            # When this happens, the c code instead surfaces the return and reason
+            # codes which causes a DownstreamFatalError to be raised.
+            raise DownstreamFatalError(
+                result_xml, request_xml, self.get_running_userid()
+            )
         if self.__debug:
             # No need to redact anything here since the raw result XML
             # already has secrets redacted when it is built.
             self.__logger.log_xml("Result XML", result_xml)
-        results = SecurityResult(result_xml)
+        results = SecurityResult(result_xml, self.get_running_userid())
         if self.__debug:
             # No need to redact anything here since the result dictionary
             # already has secrets redacted when it is built.
             self.__logger.log_dictionary(
                 "Result Dictionary", results.get_result_dictionary()
             )
         result_dictionary = results.get_result_dictionary()
+        if result_dictionary["securityResult"]["returnCode"] > 4:
```

Comment:
```python
if result_dictionary["securityResult"]["returnCode"] == 8:
```

Reply:
this actually isn't quite the case, I was mistaken. This return code is NOT the SAF return code as that isn't always validly read into our results dictionary. I can make this >=8 or >4 as initially specified. IRRSMO00 documents that the RACF return code is only greater than 4 when the SAF return code is also, but the RACF return code can be LOTS of values in that range.

Reply:
Ok, if a return code of anything 8 or greater is indicative of a problem with IRRSMO00, then I think a conditional of `>=8` makes sense given that we could assume that any return code lower than 8 would be indicative of a request that processed successfully.

Reply:
Sure, change made.

Reply:
This conversation is still unresolved.

Reply:
Sorry, this got lumped in and hidden with a bunch of conversations that were resolved on my view. Thank you for bringing my attention back to it.

I can easily change this comment to "greater than or equal to 8" to reflect the code more directly. These are all indicative of an issue with IRRSMO00 that would stop a command image from being generated.

The comment now reads:

```python
            # All return codes greater than or equal to 8 are indicative of an issue
            # with IRRSMO00 that would stop RACF from generating a command image.
            # The user should interogate the result dictionary attached to the
            # SecurityRequestError as well as the return and reason codes to resolve
            # the problem.
```